lib/index.js

```diff
@@ -0,0 +1,37 @@
+module.exports = throttlingPlugin
+
+const Bottleneck = require('bottleneck/light')
+const ROUTES = require('@octokit/routes')
```

Comment:
we cannot require `@octokit/routes` directly unfortunately, it’s huuuuge. Instead I usually have a build script that parses `@octokit/routes` and creates a new file with only the information we need.

The setup is as follows

1. Install `@octokit/routes` as dev dependency, pin the version
2. Create a new script like `scripts/generate-routes-triggering-notifications.js` that will create/update a file such as `lib/routes-triggering-notifications.json`. The script would require `@octokit/routes`, look for endpoints with the `triggersNotification` flag and then return an array with only the URLs, for example.
3. Setup travis & install [Greenkeeper](https://greenkeeper.io/)
4. In `.travis.yml`, add a "generate-routes-triggering-notifications" stage that runs the above script, that only gets run if a branch name matches `^greenkeeper/@octokit/routes`. After the script, push possible changes to the same branch so the pull request by GitHub gets updated

You can see that setup in action in the SCIM plugin, for example: https://github.com/octokit/plugin-scim.js. Here is one of the PRs that Greenkeeper created: https://github.com/octokit/plugin-scim.js/pull/16. There is no additional commit because there was no relevant change. Here is another PR that had changes: https://github.com/octokit/plugin-scim.js/pull/4

Hope that all makes sense :)

Reply:
Absolutely, this plugin will only have one dependency: Bottleneck. I looked into the octokit/routes repo a few days ago and saw all the pre-built stuff. I'll try to follow your setup.

Reply:
The GitHub token for the "push possible changes to the same branch so the pull request by GitHub gets updated" will be set on Travis as environment variable once I setup semantic-release for this repository. I’ll do that right before merging the PR